core/services/job/orm.go

```diff
@@ -252,18 +253,60 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 					if !useForwarders && err != nil {
 						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
 					}
+
+					chainIDInterface, ok := jb.OCR2OracleSpec.RelayConfig["chainID"]
+					if !ok {
+						return errors.Errorf("missing chainID in EVM type in OCR2 job spec")
+					}
+					newChainID, ok := chainIDInterface.(int64)
+					if !ok {
+						return errors.Errorf("failed to parse chainID in OCR2 job spec")
+					}
+
+					var specs []OCR2OracleSpec
+					err = tx.Select(&specs, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2`,
+						"EVM", jb.OCR2OracleSpec.ContractID,
+					)
+
+					if !errors.Is(err, sql.ErrNoRows) {
+						if err != nil {
+							return errors.Wrapf(err, "Unexpected error validating contract_id")
+						}
+						for _, spec := range specs {
+							chainIDValue, ok := spec.RelayConfig["ChainID"]
+							if !ok {
+								return errors.Errorf("Unexpected error validating contract_id")
+							}
+							chainID, ok := chainIDValue.(int64)
+							if !ok {
+								return errors.Errorf("Unexpected error validating contract_id")
+							}
```

Comment:
These errors should say something about what happened:
```suggestion
							chainIDValue, ok := spec.RelayConfig["chainID"]
							if !ok {
								return errors.Errorf("Unexpected error validating contract_id")
							}
							chainID, ok := chainIDValue.(int64)
							if !ok {
								return errors.Errorf("Unexpected error validating contract_id")
							}
```
Or, maybe this repeated block could be wrapped up in a helper func/method to reduce noise and copy/paste errors, then we can just wrap it here.